chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -525,47 +526,81 @@ func (b *BitcoindNotifier) confDetailsManually(confRequest chainntnfs.ConfReques
 	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation,
 	chainntnfs.TxConfStatus, error) {
 
-	// Begin scanning blocks at every height to determine where the
-	// transaction was included in.
+	// batchRequest will store the scan requests at every height to determine
+	// where the transaction was included in.
+	batchRequest := make(
+		map[uint32]rpcclient.FutureGetBlockHashResult, currentHeight-heightHint,
+	)
+
 	for height := currentHeight; height >= heightHint && height > 0; height-- {
+		//BatchClient works with AsynRequest.
+		sendReq, err := b.chainConn.GetBlockHashAsync(int64(height))
+		if err != nil {
+			return nil, chainntnfs.TxNotFoundManually, fmt.Errorf("unable to "+
+				"receive the block promise %v", err)
+		}
+
+		batchRequest[height] = sendReq
+	}
+
+	// Sending the bulk request using the updated batchClient.
+	b.chainConn.SendAsyncQueue()
+
+	blockHashAddresses := make([]*chainhash.Hash, 0, len(batchRequest))
+
+	for height, response := range batchRequest {
 		// Ensure we haven't been requested to shut down before
-		// processing the next height.
+		// processing the next height response.
 		select {
 		case <-b.quit:
 			return nil, chainntnfs.TxNotFoundManually,
 				chainntnfs.ErrChainNotifierShuttingDown
 		default:
 		}
 
-		blockHash, err := b.chainConn.GetBlockHash(int64(height))
+		// First, we'll fetch the block for the current height.
+		blockHash, err := response.Receive()
 		if err != nil {
-			return nil, chainntnfs.TxNotFoundManually,
-				fmt.Errorf("unable to get hash from block "+
-					"with height %d", height)
+			return nil, chainntnfs.TxNotFoundManually, fmt.Errorf("unable to "+
+				"retrieve hash for block with height %d: %v", height, err)
 		}
 
-		block, err := b.GetBlock(blockHash)
-		if err != nil {
+		blockHashAddresses = append(blockHashAddresses, blockHash)
+	}
+
+	blocks, err := b.chainConn.GetBlocksBatch(blockHashAddresses)
```

Comment:
It is very cool that we can now use `GetBlocksBatch` here! One thing that could break LND here is if we were to batch get say 1000 blocks. In that case low memory devices may OOM and crash LND. I think now that the main batching logic is there, you could do batch requests to get 10-25 blocks at most. This would help avoiding hogging too much memory as well as it'd be faster to quit as we'd check on the `quit` channel more often.